server/src/main/java/com/exacaster/lighter/application/sessions/SessionHandler.java

```diff
@@ -49,10 +52,37 @@ public void launch(Application application, Consumer<Throwable> errorHandler) {
         app.launch(backend.getSubmitConfiguration(application));
     }
 
+    @EventListener
+    @Async
+    public void startPermanentSession(StartupEvent event) {
+        sessionService.fetchPermanent().ifPresent(it -> sessionService.deleteOne(it.getId()));
+        startPermanentSession();
+    }
+
+    private void startPermanentSession() {
+        var params = appConfiguration.getSessionConfiguration().getPermanentSessionParams();
+        if (params != null) {
+            var session = sessionService.createSession(params, ApplicationType.PERMANENT_SESSION);
+            statusTracker.processApplicationStarting(session);
+            launch(session, error -> statusTracker.processApplicationError(session, error));
+        }
+    }
+
+    @SchedulerLock(name = "keepPermanentSession")
+    @Scheduled(fixedRate = "1m")
+    public void keepPermanentSession() {
```

Comment:
We should not timeout permanent sessions instead of restarting.

Reply:
Permanent sessions are not timed out. Only completed(ERROR, DEAD, KILLED, SUCCESS) permanent sessions are restarted.